x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/time_range_metadata/many_apm_server_versions.spec.ts

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+import moment from 'moment';
+import { ApmSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import {
+  TRANSACTION_DURATION_HISTOGRAM,
+  TRANSACTION_DURATION_SUMMARY,
+} from '@kbn/apm-plugin/common/es_fields/apm';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { LatencyAggregationType } from '@kbn/apm-plugin/common/latency_aggregation_types';
+import { Readable } from 'stream';
+import type { ApmApiClient } from '../../../../services/apm_api';
+import type { DeploymentAgnosticFtrProviderContext } from '../../../../ftr_provider_context';
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const apmApiClient = getService('apmApi');
+  const synthtrace = getService('synthtrace');
+  const es = getService('es');
+
+  const baseTime = new Date('2023-10-01T00:00:00.000Z').getTime();
+  const startLegacy = moment(baseTime).add(0, 'minutes');
+  const start = moment(baseTime).add(5, 'minutes');
+  const endLegacy = moment(baseTime).add(10, 'minutes');
+  const end = moment(baseTime).add(15, 'minutes');
+
+  describe('Time range metadata when there are multiple APM Server versions', () => {
+    describe('when ingesting traces from APM Server with different versions', () => {
+      let apmSynthtraceEsClient: ApmSynthtraceEsClient;
+
+      before(async () => {
+        apmSynthtraceEsClient = await synthtrace.createApmSynthtraceEsClient();
+        await generateTraceDataForService({
+          serviceName: 'synth-java-legacy',
+          start: startLegacy,
+          end: endLegacy,
+          isLegacy: true,
+          synthtrace: apmSynthtraceEsClient,
+        });
+
+        await generateTraceDataForService({
+          serviceName: 'synth-java',
+          start,
+          end,
+          isLegacy: false,
+          synthtrace: apmSynthtraceEsClient,
+        });
+      });
+
+      after(() => {
+        return apmSynthtraceEsClient.clean();
+      });
+
+      it('ingests transaction metrics with transaction.duration.summary', async () => {
+        const res = await es.search({
+          index: 'metrics-apm*',
+          body: {
+            query: {
+              bool: {
+                filter: [
+                  { exists: { field: TRANSACTION_DURATION_HISTOGRAM } },
+                  { exists: { field: TRANSACTION_DURATION_SUMMARY } },
+                ],
+              },
+            },
+          },
+        });
```

Comment:
FYI: currently we init ES client with `operator` user as we use it to cleanup MKI projects. There is no way to change authentication for `es` service, so if your test target validation from end-user perspective you need to make a change and probably use a new ES client instance with API key authentication (I didn't see it in tests before, but I assume it should work)

Reply:
Thanks for sharing that. This validation here is just ensuring `es` has ingested the necessary data to proceed with the tests